src/communication/index.ts

```diff
@@ -99,6 +102,21 @@ export async function broadcast (code: MessageCodesEnum, payload: Record<string,
 
   logger.verbose(`Broadcasting message with code: ${code}`)
 
-  // TODO: Persist the sent messages for "rebroadcast"
+  await Message.create({
+    code,
+    agreementReference: payload.agreementReference,
+    message: JSON.stringify(msg)
+  })
+  const messageCountForAgreement = await Message.count({ where: { agreementReference: payload.agreementReference } })
+
+  if (messageCountForAgreement > config.get<number>('comms.countOfMessagesPersistedPerAgreement')) {
+    // We have reached maximum messages for agreement
+    // We have added one Message, so we gonna remove the oldest one
+    const latestMessage = await Message.findOne({
+      where: { agreementReference: payload.agreementReference }
+    })
+    await latestMessage?.destroy()
```

Comment:
Hmm, should we remove all messages which is out of limit and not only the last message?

Reply:
Clear

Reply:
But from other hands, message broadcasting can be called asynchronously. That's mean that at the same time you can create two messages. If this will work in that case?

Reply:
Sorry, forgot to come back to this 😅

Hmmm, I guess it would be better to do a more robust solution, you are right.

Reply:
Cool, thank you ;)